packages/react-router/lib/hooks.tsx

```diff
@@ -311,14 +309,7 @@ export function useResolvedPath(
 ): Path {
   let { matches } = React.useContext(RouteContext);
   let { pathname: locationPathname } = useLocation();
-
-  // Use the full pathname for the leaf match so we include splat values
-  // for "." links
-  let routePathnamesJson = JSON.stringify(
-    getPathContributingMatches(matches).map((match, idx) =>
-      idx === matches.length - 1 ? match.pathname : match.pathnameBase
-    )
-  );
+  let routePathnamesJson = JSON.stringify(getResolveToMatches(matches));
```

Comment:
This change on line 319 was only made in the one code path before.  To avoid this same mistake in the future, we're now using a utility `getResolveToMatches` to share this logic.